core/src/main/java/org/apache/iceberg/rest/RESTViewOperations.java

```diff
@@ -59,6 +60,8 @@ public void commit(ViewMetadata base, ViewMetadata metadata) {
     // this is only used for replacing view metadata
     Preconditions.checkState(base != null, "Invalid base metadata: null");
 
+    ViewUtil.verifyReplaceStrictness(base, metadata);
```

Comment:
What's the rationale for having this in the `ViewUtil` and in a public API? I think it makes more sense to be `protected` within `ViewOperations` so that different implementations can just call out to it during the commit. I don't think it needs to be exposed more than that at least at this point in time.

Reply:
see https://github.com/apache/iceberg/pull/9620#discussion_r1478246464